libs/@blockprotocol/type-system/typescript/src/native/url.ts

```diff
@@ -106,34 +107,94 @@ export const validateVersionedUrl = (
       };
     }
 
-    const index = version.search(/[^0-9]/);
-    if (index === 0) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "InvalidVersion",
-          inner: [version, "invalid digit found in string"],
-        },
-      };
-    } else if (index > 0) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "AdditionalEndContent",
-          inner: version.substring(index),
-        },
-      };
-    }
-
-    const versionNumber = Number(version);
-    if (versionNumber > 4294967295) {
-      return {
-        type: "Err",
-        inner: {
-          reason: "InvalidVersion",
-          inner: [version, "number too large to fit in target type"],
-        },
-      };
+    // Try to match draft version format first: "2-draft.lane.5"
+    const draftMatch = version.match(/^(\d+)-draft\.([^.]+)\.(\d+)$/);
+
+    if (draftMatch) {
+      // It's a draft version
+      const [, majorStr, _lane, revisionStr] = draftMatch;
+      const major = Number(majorStr);
+      const revision = Number(revisionStr);
+
+      if (major > 4294967295) {
```

Comment:
Did you mean to use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER